packages/client/src/store/actions.ts

```diff
@@ -119,3 +152,94 @@ export function unregisterAbility( name: string ) {
 		name,
 	};
 }
+
+/**
+ * Registers a client-side ability category in the store.
+ *
+ * This action validates the category before registration. If validation fails,
+ * an error will be thrown. Categories will be automatically fetched from the
+ * REST API if they haven't been loaded yet to check for duplicates.
+ *
+ * @param  slug The unique category slug identifier.
+ * @param  args Category arguments (label, description, optional meta).
+ * @return Action object or function.
+ * @throws {Error} If validation fails.
+ */
+export function registerAbilityCategory(
+	slug: string,
+	args: AbilityCategoryArgs
+) {
+	// @ts-expect-error - registry types are not yet available
+	return async ( { select, dispatch } ) => {
+		if ( ! slug ) {
+			throw new Error( 'Category slug is required' );
+		}
+
+		// Validate slug format matches server implementation
+		if ( ! /^[a-z0-9]+(?:-[a-z0-9]+)*$/.test( slug ) ) {
+			throw new Error(
+				'Category slug must contain only lowercase alphanumeric characters and dashes.'
+			);
+		}
+
+		// Ensure categories are loaded before checking for duplicates
+		await resolveSelect( STORE_NAME ).getAbilityCategories();
+		const existingCategory = select.getAbilityCategory( slug );
+		if ( existingCategory ) {
+			throw new Error(
+				sprintf( 'Category "%s" is already registered.', slug )
+			);
+		}
```

Comment:
@emdashcodes Just double-checking: It's this `await resolveSelect` here that ensures the server categories are loaded prior to checking for a collision, right?